gazebo/physics/ode/ODEPhysics.cc

```diff
@@ -1161,6 +1164,111 @@ void ODEPhysics::Collide(ODECollision *_collision1, ODECollision *_collision2,
     contact.fdir1[0] = fd.X();
     contact.fdir1[1] = fd.Y();
     contact.fdir1[2] = fd.Z();
+
+    // Check for plowing parameters if friction direction is specified
+    auto collision1Sdf = _collision1->GetSDF();
+    auto collision2Sdf = _collision2->GetSDF();
+
+    const std::string kPlowingTerrain =
+      "gz:plowing_terrain";
+    const std::string kPlowingWheel =
+      "gz:plowing_wheel";
+
+    sdf::ElementPtr wheelSdf = nullptr;
+    sdf::ElementPtr terrainSdf = nullptr;
+
+    if (collision1Sdf->HasElement(kPlowingWheel) &&
+        collision2Sdf->HasElement(kPlowingTerrain))
+    {
+      wheelSdf = collision1Sdf->GetElement(kPlowingWheel);
+      terrainSdf = collision2Sdf;
+    }
+    else if (collision2Sdf->HasElement(kPlowingWheel) &&
+             collision1Sdf->HasElement(kPlowingTerrain))
+    {
+      wheelSdf = collision2Sdf->GetElement(kPlowingWheel);
+      terrainSdf = collision1Sdf;
+    }
+
+    const std::string kPlowingMaxDegrees =
+      "max_degrees";
+    const std::string kPlowingSaturationVelocity =
+      "saturation_velocity";
+    if (wheelSdf && terrainSdf &&
+        wheelSdf->HasElement(kPlowingMaxDegrees) &&
+        wheelSdf->HasElement(kPlowingSaturationVelocity))
+    {
+      const std::string kPlowingDeadbandVelocity =
+        "deadband_velocity";
+      ignition::math::Angle plowingMaxAngle;
+      plowingMaxAngle.SetDegree(wheelSdf->Get<double>(kPlowingMaxDegrees));
+      double plowingSaturationVelocity =
+        wheelSdf->Get<double>(kPlowingSaturationVelocity);
+      // Use deadband velocity = 0 if parameter is not set
+      double plowingDeadbandVelocity =
+        wheelSdf->Get<double>(kPlowingDeadbandVelocity, 0.0).first;
+
+      // compute slope
+      double slope = plowingMaxAngle.Radian() /
+          (plowingSaturationVelocity - plowingDeadbandVelocity);
```

Comment:
check that saturation velocity is larger than deadband velocity

Reply:
this ended up being a bigger change, as I moved the SDF parameter parsing to a static helper function in ODECollision and added parameter checks there

https://github.com/osrf/gazebo/pull/3164/commits/dd64a81cc396cc1ef7adf27fd167867bb0bc1570